src/modules/share/share.repository.ts

```diff
@@ -0,0 +1,257 @@
+  Default,
+  Unique,
+  AllowNull,
+} from 'sequelize-typescript';
+import { FileModel } from '../file/file.repository';
+import { User, UserAttributes } from '../user/user.domain';
+import { UserModel } from '../user/user.repository';
+import { FolderModel } from '../folder/folder.repository';
+import { Folder, FolderAttributes } from '../folder/folder.domain';
+import { Pagination } from '../../lib/pagination';
+@Table({
+  underscored: true,
+  timestamps: true,
+  tableName: 'shares',
+})
+export class ShareModel extends Model {
+  @PrimaryKey
+  @Column
+  id: number;
+
+  @Unique
+  @Column
+  token: string;
+
+  @Column(DataType.BLOB)
+  mnemonic: string;
+
+  @ForeignKey(() => UserModel)
+  @Column
+  userId: number;
+
+  @BelongsTo(() => UserModel)
+  user: UserModel;
+
+  @ForeignKey(() => FileModel)
+  @Column
+  fileId: number;
+
+  @BelongsTo(() => FileModel, 'fileId')
+  file: FileModel;
+
+  @ForeignKey(() => FolderModel)
+  @Column
+  folderId: number;
+
+  @BelongsTo(() => FolderModel, 'folderId')
+  folder: FolderModel;
+
+  @Column(DataType.STRING(64))
+  encryptionKey: string;
+
+  @Column(DataType.STRING(24))
+  bucket: string;
+
+  @Column(DataType.STRING(64))
+  fileToken: string;
+
+  @Default(false)
+  @Column
+  isFolder: boolean;
+
+  @Default(1)
+  @Column
+  views: number;
+
+  @AllowNull
+  @Column
+  timesValid: number;
+
+  @Default(true)
+  @Column
+  active: boolean;
+}
+
+export interface ShareRepository {
+  findById(id: number): Promise<Share | null>;
+  findByToken(token: string): Promise<Share | null>;
+  findAllByUserPaginated(
+    user: any,
+    page: number,
+    perPage: number,
+  ): Promise<{ count: number; items: Array<Share> | [] }>;
+  update(share: Share): Promise<true>;
+  create(share: Share): Promise<void>;
+  findByFileIdAndUser(
+    fileId: FileAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<Share>;
+  findByFolderIdAndUser(
+    folderId: FolderAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<Share>;
+}
+
+@Injectable()
+export class SequelizeShareRepository implements ShareRepository {
+  constructor(
+    @InjectModel(ShareModel)
+    private shareModel: typeof ShareModel,
+    @InjectModel(FileModel)
+    private fileModel: typeof FileModel,
+    @InjectModel(FolderModel)
+    private folderModel: typeof FolderModel,
+    @InjectModel(UserModel)
+    private userModel: typeof UserModel,
+  ) {}
+
+  async findById(id: ShareAttributes['id']) {
+    const share = await this.shareModel.findByPk(id, {
+      include: [this.fileModel, this.userModel, this.folderModel],
+    });
+    return share ? this.toDomain(share) : null;
+  }
+
+  async findByFileIdAndUser(
+    fileId: FileAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<Share> {
+    const share = await this.shareModel.findOne({
+      where: { fileId, userId },
+      include: [this.fileModel, this.userModel],
+    });
+    return share ? this.toDomain(share) : null;
```

Comment:
Same here, if the return could be null or a Share, please, type it properly. Also, the repository interface does it for you.